gnomad/utils/sparse_mt.py

```diff
@@ -917,29 +917,56 @@ def compute_coverage_stats(
     It needs to be keyed with the same keys as `mt`, typically either `locus` or `locus, alleles`.
     The `reference_ht` can e.g. be created using `get_reference_ht`
 
-    :param mt: Input sparse MT
+    :param mtds: Input sparse MT or VDS
     :param reference_ht: Input reference HT
+    :param interval_ht: Table containing intervals to filter to
     :param coverage_over_x_bins: List of boundaries for computing samples over X
     :return: Table with per-base coverage stats
     """
-    n_samples = mt.count_cols()
-    print(f"Computing coverage stats on {n_samples} samples.")
+    is_vds = isinstance(mtds, hl.vds.VariantDataset)
+    if is_vds:
+        n_samples = mtds.variant_data.count_cols()
+    else:
+        n_samples = mtds.count_cols()
+    logging.info(f"Computing coverage stats on {n_samples} samples.")
+
+    # Filter to interval list
+    if interval_ht is not None:
+        if is_vds:
+            mtds = hl.vds.filter_intervals(
+                vds=mtds, intervals=interval_ht, split_reference_blocks=True
+            )
+        else:
+            mtds = hl.filter_intervals(mtds, interval_ht["interval"].collect())
 
     # Create an outer join with the reference Table
-    mt = mt.select_entries("END", "DP").select_cols().select_rows()
-    col_key_fields = list(mt.col_key)
-    t = mt._localize_entries("__entries", "__cols")
-    t = t.join(reference_ht.key_by(*mt.row_key).select(_in_ref=True), how="outer")
-    t = t.annotate(
-        __entries=hl.or_else(
-            t.__entries,
-            hl.range(n_samples).map(lambda x: hl.null(t.__entries.dtype.element_type)),
+    def join_with_ref(mt: hl.MatrixTable) -> hl.MatrixTable:
```

Comment:
added, although now wondering, do we need this step for exomes?

Reply:
wow, sorry, thank you for saying this because it made me realize I think you want to filter the reference HT by the interval list, not the `mtds`, but then yes I do think we want it after that